### DISCUSSION TITLE
Key error

### DISCUSSION THREAD

**Discussion by allenmoench** [earthlab-education/Earth-Analytics-AY24#561] [Q&A]
Strange thing happened. I've been working on the Species distribution notebooks, and errors started appearing in sections of code that had run smoothly before. I think it's basically saying that eco_gdf is not definied. Here's what it looks like: 
![image](https://github.com/user-attachments/assets/ca2dade1-f271-4ed4-a3c4-8ef351e265b9)
down at the bottom of the error message, it says KeyError: "['index_right'] not in index".

Anyone know what's going on here?

**Comment by hanried** [comment]
I think that means that neither your eco_gdf nor your gbif_gdf have a column called 'index_right'. 

Playing around in my own code, instead of selecting the 'month' column, I selected the 'mon' column, which doesn't exist in either of my gdfs, and got the same error language.

<img width="625" alt="image" src="https://github.com/user-attachments/assets/8233cbed-5122-4240-9d8b-1d300d22f28f">

![image](https://github.com/user-attachments/assets/32b90257-ff12-4018-bf57-ce5dd6f3ca3a)

**Comment by hanried** [comment]
Also, if it helps, I think you can check what the index and keys (aka columns, I think) of your gdfs are with this:

gdf.keys()
gdf.index

![image](https://github.com/user-attachments/assets/6b0356e5-cb8d-4354-8891-fe41764a7d41)

**Comment by eculler** [comment]
@allenmoench -- I agree with @hanried, it looks like it's complaining about the column. I had this problem when I did the videos, and I'm honestly not sure where it came from. However, I did go through and fix the earlier code in the second video, so you could check that out. I didn't want to redo the videos for you all because I wanted folks who had already done the first video to be able to continue with the second.
